src/registrar/models/domain.py

```diff
@@ -381,78 +719,226 @@ def _validate_host_tuples(self, hosts: list[tuple[str]]):
                 else:
+                    logger.error(e)
+                    logger.error(e.code)
                     raise e
 
+    def addRegistrant(self):
+        registrant = PublicContact.get_default_registrant()
+        registrant.domain = self
+        registrant.save()  # calls the registrant_contact.setter
+        return registrant.registry_id
+
+    @transition(field="state", source=State.UNKNOWN, target=State.DNS_NEEDED)
+    def pendingCreate(self):
+        logger.info("Changing to dns_needed")
+
+        registrantID = self.addRegistrant()
+
+        # TODO-notes no chg item for registrant in the epplib should
+
+        req = commands.CreateDomain(
+            name=self.name,
+            registrant=registrantID,
+            auth_info=epp.DomainAuthInfo(pw="2fooBAR123fooBaz"),  # not a password
+        )
+
+        try:
+            registry.send(req, cleaned=True)
+
+        except RegistryError as err:
+            if err.code != ErrorCode.OBJECT_EXISTS:
+                raise err
+
+        self.addAllDefaults()
+
+    def addAllDefaults(self):
+        security_contact = self.get_default_security_contact()
+        security_contact.save()
+
+        technical_contact = PublicContact.get_default_technical()
+        technical_contact.domain = self
+        technical_contact.save()
+
+        administrative_contact = PublicContact.get_default_administrative()
+        administrative_contact.domain = self
+        administrative_contact.save()
+
+    @transition(field="state", source=State.DNS_NEEDED, target=State.ON_HOLD)
+    def clientHold(self):
+        """place a clienthold on a domain (no longer should resolve)"""
+        # TODO - ensure all requirements for client hold are made here
+        # (check prohibited statuses)
+        logger.info("clientHold()-> inside clientHold")
+        self._place_client_hold()
+        # TODO -on the client hold ticket any additional error handling here
+
+    @transition(field="state", source=State.ON_HOLD, target=State.DNS_NEEDED)
+    def revertClientHold(self):
+        """undo a clienthold placed on a domain"""
+
+        logger.info("clientHold()-> inside clientHold")
+        self._remove_client_hold()
+        # TODO -on the client hold ticket any additional error handling here
+
+    @transition(field="state", source=State.ON_HOLD, target=State.DELETED)
+    def deleted(self):
+        """domain is deleted in epp but is saved in our database"""
+        # TODO Domains may not be deleted if:
+        #  a child host is being used by
+        # another .gov domains.  The host must be first removed
+        # and/or renamed before the parent domain may be deleted.
+        logger.info("pendingCreate()-> inside pending create")
+        self._delete_domain()
+        # TODO - delete ticket any additional error handling here
+
+    @transition(
+        field="state",
+        source=[State.DNS_NEEDED],
+        target=State.READY,
+    )
+    def ready(self):
```

Comment:
consider adding the contents of the if as a condition for transitioning. check how this effects error handling and messaging for the user.